### PR TITLE
updates k8s ingress resource to generate the current non-deprecated format

### PR DESCRIPTION
**Pull request by xetys**
Since Kubernetes version 1.22 the `Ingress` resource version `v1beta1` is deprecated and no longer supported.

Using the old version prevents users of k8s version 1.22+ to deploy our generated k8s manifests

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: test/__snapshots__/kubernetes.helm.spec.js.snap [hunk 1]
```diff
@@ -1609,20 +1609,24 @@ spec:
...
   },
   "jhgate-helm/templates/jhgate-ingress.yml": Object {
-    "contents": "apiVersion: networking.k8s.io/v1beta1
+    "contents": "apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: jhgate
   namespace: default
 spec:
+  ingressClassName: nginc
```
Comment (DanielFran) [inline test/__snapshots__/kubernetes.helm.spec.js.snap:1618]:
```suggestion
  ingressClassName: nginx
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/kubernetes/templates/ingress.yml.ejs [hunk 2]
```diff
@@ -22,6 +22,7 @@ metadata:
   name: <%= app.baseName.toLowerCase() %>
   namespace: <%= kubernetesNamespace %>
 spec:
+  ingressClassName: <%= ingressTypeNginx || "nginx" %>
```
Comment (mshima) [inline generators/kubernetes/templates/ingress.yml.ejs:25]: Probably you mean this
```suggestion
  ingressClassName: <%= ingressType || "nginx" %>
```

Comment (xetys) [reply generators/kubernetes/templates/ingress.yml.ejs:25]: :see_no_evil: it's been so long....thanks for seeing my silly mistakes >,<

**Comment by mshima** [comment]
@xetys trusting you

**Comment by mshima** [comment]
Thanks

**Pull request merged**